osv/purl_helpers.py

```diff
@@ -63,7 +66,84 @@ def package_to_purl(ecosystem: str, package_name: str) -> str | None:
   return f'pkg:{purl_type}/{_url_encode(package_name)}{suffix}'
 
 
-def purl_to_ecosystem(purl_type: str) -> str | None:
-  """Convert purl to a specific ecosystem string"""
-  ecosystem_purl = {v: k for k, v in PURL_ECOSYSTEMS.items()}
-  return ecosystem_purl.get(purl_type)
+def parse_purl(purl_str: str) -> tuple[str, str, str] | None:
+  """Parses a PURL string and extracts
+  ecosystem, package, and version information.
+
+  Args:
+    purl_str: The Package URL string to parse.
+
+  Returns:
+    A tuple containing the ecosystem, package,
+    and version, or None if parsing fails.
+  """
+
+  try:
+    purl = PackageURL.from_string(purl_str)
+  except ValueError:  # Catch potential parsing errors
+    return None
+
+  match purl:
+    case PackageURL(type='bitnami'):
+      ecosystem = 'Bitnami'
+    case PackageURL(type='cargo'):
+      ecosystem = 'crates.io'
+    case PackageURL(
+        type='golang', namespace=namespace, name=name, version=version):
+      # OSV uses the combined purl.namespace and purl.name for Go package names
```

Comment:
Nit: I think it's not so much OSV as it is Go itself?

```suggestion
      # Go uses the combined purl.namespace and purl.name for Go package names
```